core/app/jobs/spree/themes/screenshot_job.rb

```
@@ -0,0 +1,70 @@
+      SCREENSHOT_OUTPUT = 'image'.freeze
+      SCREENSHOT_FILE_TYPE = 'png'.freeze
+
+      def perform(theme_id)
+        theme = Spree::Theme.find(theme_id)
+
+        Rails.logger.warn('Screenshot API token is missing') and return if screenshot_api_token.blank?
+
+        return if theme.screenshot.attached?
+
+        url = URI.encode_www_form_component("#{theme.store.url_or_custom_domain}?theme_id=#{theme_id}")
+        query = "#{BASE_SCREENSHOT_API_URL}?token=#{screenshot_api_token}&url=#{url}&output=#{SCREENSHOT_OUTPUT}&file_type=#{SCREENSHOT_FILE_TYPE}&retina=true&enable_caching=true"
```

RuboCop (1.73)

[convention] Line is too long. [179/150]

(Layout/LineLength)